be/src/storage/task/engine_checksum_task.cpp

```diff
@@ -64,69 +63,50 @@ OLAPStatus EngineChecksumTask::_compute_checksum() {
         return OLAP_SUCCESS;
     }
 
-    Reader reader;
-    ReaderParams reader_params;
-    reader_params.tablet = tablet;
-    reader_params.reader_type = READER_CHECKSUM;
-    reader_params.version = Version(0, _version);
-
-    {
-        std::shared_lock rdlock(tablet->get_header_lock());
-        const RowsetSharedPtr message = tablet->rowset_with_max_version();
-        if (message == nullptr) {
-            LOG(FATAL) << "fail to get latest version. tablet_id=" << _tablet_id;
-            return OLAP_ERR_VERSION_NOT_EXIST;
-        }
-
-        if (Status st = tablet->capture_rs_readers(reader_params.version, &reader_params.rs_readers); !st.ok()) {
-            LOG(WARNING) << "fail to init reader for tablet " << tablet->full_name() << ": " << st;
-            return OLAP_ERR_OTHER_ERROR;
+    std::vector<uint32_t> return_columns;
+    const TabletSchema& tablet_schema = tablet->tablet_schema();
+    for (size_t i = 0; i < tablet_schema.num_columns(); ++i) {
+        FieldType type = tablet_schema.column(i).type();
+        // The approximation of FLOAT/DOUBLE in a certain precision range, the binary of byte is not
+        // a fixed value, so these two types are ignored in calculating checksum.
+        // And also HLL/OBJCET/PERCENTILE is too large to calculate the checksum.
+        if (type == OLAP_FIELD_TYPE_FLOAT || type == OLAP_FIELD_TYPE_DOUBLE || type == OLAP_FIELD_TYPE_HLL ||
+            type == OLAP_FIELD_TYPE_OBJECT || type == OLAP_FIELD_TYPE_PERCENTILE) {
+            continue;
         }
+        return_columns.push_back(i);
     }
 
-    for (size_t i = 0; i < tablet->tablet_schema().num_columns(); ++i) {
-        reader_params.return_columns.push_back(i);
-    }
+    vectorized::Schema child_schema =
+            vectorized::ChunkHelper::convert_schema_to_format_v2(tablet_schema, return_columns);
 
-    res = reader.init(reader_params);
-    if (res != OLAP_SUCCESS) {
-        LOG(WARNING) << "initiate reader fail. res=" << res;
-        return res;
+    std::shared_ptr<vectorized::Reader> reader =
+            std::make_shared<vectorized::Reader>(tablet, Version(0, _version), std::move(child_schema));
```

Comment:
`vectorized::Reader` has been renamed to `vectorized::TabletReader` in the main branch.